mountpoint-s3/src/inode.rs

```diff
@@ -515,7 +553,10 @@ pub struct ReaddirHandle {
     parent_ino: InodeNo,
     full_path: String,
     page_size: usize,
-    results: RwLock<VecDeque<LookedUp>>,
+    // Use binary heap as a priority queue
+    results: RwLock<BinaryHeap<LookedUp>>,
+    // A set of files that only exist locally, we use them for deduplication
+    local_results: RwLock<HashSet<InodeNo>>,
```

Comment:
Just a suggestion of a different approach: instead of a priority queue plus a set, we just have two `VecDeque`s, one "remote" and one "local". Then `ReaddirHandle::next` effectively merge-sorts them—refills the "remote" queue if it's empty, and then compares the first element in both lists to decide which to return next, breaking the tie if necessary.

Not sure which of these approaches is clearer, though? My intuition here is that the priority queue isn't quite what we want because entries never get inserted anywhere except the bottom, but perhaps it's worth the trouble anyway.